### ISSUE TITLE
Example of one dropdown field leading to updated values for another dropdown field

### ISSUE THREAD

**Issue by dukedougal** [erikras/redux-form#624]
I have a container component and a page component. The page component is a redux form. The form is stateless.

My goal is to have two dropdown select fields, changing the first select results in a new set of values in the second select & both dropdown select fields load their values via ajax.

I have made this work but with a great deal of code and I'm sure I've made it too complex.  Posting the code here would be a huge exercise in unwinding it from the surrounding application.

Would someone be willing to post the simplest possible implementation of the above scenario using redux form?

thanks

**Comment by erikras** [comment]
**TL;DR**: The lists of options are props from a reducer, and whenever the first dropdown changes, you clear the value and load new options for the second one.

I'm going to use a countries-cities relationship here in the example.

### Normalize to clear the cities value
```js
import {createStore, combineReducers, compose } from 'redux';
import {reducer as form} from 'redux-form';
import locationLoader from './locationLoader';

const reducer = combineReducers({
  locationLoader, // reducer in charge of loading locations
  form: form.normalize({
    myForm: {
      city: (value, previousValue, allValues, previousAllValues) => {
        if (allValues.country !== previousAllValues.country) {
          // country changed. clear out city value.
          return '';
        }
        return value;
      }
    }
  })
});
```

### Form
```js
import {loadCities, loadCountries} from '../reducers/locationLoader';

@reduxForm({
    form: 'myForm',
    fields: ['country', 'city']
  },
  state => ({
    countries: state.locationLoader.countries,
    cities: state.locationLoader.cities
  }),
  {
    loadCities,    // action creator to dispatch load cities action
    loadCountries  // action creator to dispatch load countries action
  }
)
class MyForm extends Component {
  componentDidMount() {
    this.props.loadCountries();
  }

  render() {
    const {cities, country, fields: {country, city}} = this.props;
    return (
      <form>
        <select {...country}
          onChange={event => {
          country.onChange(event);                   // send change to redux-form
          this.props.loadCities(event.target.value); // load cities in country selected
        }}
          disabled={!countries.length}               // wait for countries to load
        >
          <option>Select a country</option>
          {countries.map(option => <option value={option.id} key={option.id}>{option.name}</option>)}
        </select>
        <select {...city}
          disabled={!cities.length}                 // wait for cities to load
        >
          <option>Select a city</option>
          {cities.map(option => <option value={option.id} key={option.id}>{option.name}</option>)}
        </select>
      </form>
    );
  }
}
```

This is never-run pseudo code and it may contain typos. There are several ways to do this, but you requested the simplest one.

**Comment by JosephEarl** [comment]
How can I do this with v6? Fields have a normalize but this is only called when their value changes, not when another fields value has changed.

**Comment by draganSm** [comment]
My backend developer didn't listen to me :) so the keys (ids) of the cities are not the unique. I would like to reset (show the 1st option '-- Please select ... --')  the city each time the country has been changed.

The code looks like

  handleChange(event, newValue) {
    const { countries } = this.props;
    const result = countries.find(country => `${country.id}` === newValue);
    this.setState({ cities: result ? result.cities : [] });
  }

  render() {
    const { countries } = this.props;
    const { cities } = this.state;
    return (<div>
      <Field
        name="Country"
        label={'Country'}
        placeholder="Select Country"
        component={renderField} 
        type="select"
        options={countries}
        optionsKey="id"
        optionsValue="name"
        onChange={this.handleChange}
      />

      <Field
        name="city"
        label={'City'}
        placeholder="Select City"
        component={renderField}
        type="select"
        options={cities}
        optionsKey="id"
        optionsValue="name"
      />
    </div>);
  }

If I select the cities with the same id for two countries, when I change the country, the city is not changed.
